docs/contributors/release.md

```diff
@@ -1,162 +1,142 @@
 # Gutenberg Release Process
 
-This document serves as a checklist for building and releasing a new version of Gutenberg. It documents our release process, but it is helpful if you'd like to understand how Gutenberg ships release candidates and new versions.
+The Gutenberg Repository is used to perform several types of releases. This document serves as a checklist for each one of these. It is helpful if you'd like to understand the different workflows.
 
 To release Gutenberg, you need commit access to the [WordPress.org plugin repository]. 🙂
 
-## Gutenberg Versioning
+## Plugin Releases
 
-Gutenberg follows semantic versioning; we release a new major version approximately every two weeks. The current and next versions are [tracked in GitHub milestones](https://github.com/WordPress/gutenberg/milestones), along with each version's tagging date. The date in the milestone is the date of **tagging the release candidate**, not the final version that will be published to the [plugin repository].
+### Versioning
 
-Up until version `4.0.0`, "point releases"/minor releases (eg. `3.8` > `3.9`) effectively constituted a major version bump.
+We release a new major version approximately every two weeks. The current and next versions are [tracked in GitHub milestones](https://github.com/WordPress/gutenberg/milestones), along with each version's tagging date.
 
-Our [deprecation policy](./deprecated.md) maintains deprecated APIs/functions for two major versions, whenever possible.
+### Release Candidates
 
-# Release Candidates
+On the date of the current's milestone, we publish a release candidate and make it available for plugin authors and for users to test. If any bugs/regressions are found with a release candidate, a new release candidate can be published.
```

Comment:
I think "current's" should be "current". There's also an extra "for" in there that isn't needed:

```suggestion
On the date of the current milestone, we publish a release candidate and make it available for plugin authors and users to test. If any regressions are found with a release candidate, a new release candidate can be published.
```

I'd remove "bugs" and just write "regressions"; the key to shipping new RCs is usually regressions from the last release. It's not like any bug is a default candidate for a new release. 😄